pkg/util/duration.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/errors"
+)
+
+// Duration is a wrapper of time.Duration for TOML and JSON.
+type Duration string
+
+// NewDuration creates a Duration from time.Duration.
+func NewDuration(duration time.Duration) Duration {
+	return Duration(duration.String())
+}
+
+// MarshalJSON returns the duration as a JSON string.
+func (d *Duration) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, *d)), nil
+}
+
+// UnmarshalJSON parses a JSON string into the duration.
+func (d *Duration) UnmarshalJSON(text []byte) error {
+	s, err := strconv.Unquote(string(text))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	*d = Duration(s)
+	return nil
+}
+
```

Comment:
var _ toml.** = &Duration{}

--
add this make sure it impl the interface and let's reader know this.